### ISSUE TITLE
Support for enabling extensions for azurerm_postgresql_flexible_server_database

### ISSUE THREAD

**Issue by MShekow** [hashicorp/terraform-provider-azurerm#18265]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

As documented [here](https://docs.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-extensions), whoever administrates the Azure DB must explicitly put extensions on the "allow list" (so that database users can then install them via `CREATE EXTENSION ...` SQL command.

It seems that azurerm_postgresql_flexible_server_database lacks this ability to specify a list of allow-listed PostgreSQL extensions.

### New or Affected Resource(s)/Data Source(s)

azurerm_postgresql_flexible_server_database

**Comment by MShekow** [comment]
Yes, you are right, thanks for pointing it out. May I suggest to add a small hint to the fact that this [azurerm_postgresql_flexible_server_configuration](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/postgresql_flexible_server_configuration) module exists, somewhere on https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/postgresql_flexible_server_database ? And likewise to all the other modules that have a corresponding `..._configuration` module.

**Comment by goors** [comment]
`azurerm_postgresql_flexible_server_configuration` is enabling extension for server and not database. Is there a way to have this enabled for actual database? I tried `azurerm_postgresql_flexible_server_configuration` and in fact it will enable for example extensions for server but not for database.
